File: src/type/appimage.py
from typing import Optional, List, TypedDict


class Author(TypedDict):
    name: str
    url: str


class Link(TypedDict):
    type: str
    url: str


class AppImage(TypedDict):
    name: str
    description: Optional[str]
    categories: List[Optional[str]]
    authors: Optional[List[Author]]
    license: Optional[str]
    links: Optional[List[Link]]
    icons: Optional[List[str]]
    screenshots: Optional[List[str]]

File: types/appimage.go
package types

type AppImage struct {
	Version     int    "json:version"
	HomePageUrl string "json:home_page_url"
	Description string "json:description"
	Icon        string "json:icon"
	Favicon     string "json:favicon"
	Expired     bool   "json:expired"
	Items       []Item
}

type Item struct {
	Name        string "json:name"
	Description string "json:description"
	Categories  []string
	Icons       []string
	Screenshots []string
	Authors     []Author
	Links       []Link
}

type Author struct {
	Name string "json:name"
	Url  string "json:url"
}

type Link struct {
	Type string "json:type"
	Url  string "json:type"
}
